api/controlplane/kubeadm/v1beta2/kubeadmcontrolplanetemplate_types.go

```diff
@@ -46,7 +47,7 @@ type KubeadmControlPlaneTemplate struct {
 
 	// spec is the desired state of KubeadmControlPlaneTemplate.
 	// +optional
-	Spec KubeadmControlPlaneTemplateSpec `json:"spec,omitempty"`
+	Spec KubeadmControlPlaneTemplateSpec `json:"spec,omitempty,omitzero"`
```

Comment:
I think here the Spec is actually optional. See for example:  https://storage.googleapis.com/kubernetes-ci-logs/logs/periodic-cluster-api-e2e-upgrade-1-28-1-29-main/1939753255810633728/artifacts/clusters/bootstrap/resources/k8s-upgrade-and-conformance-nnfo3e/KubeadmControlPlaneTemplate/quick-start-control-plane.yaml

Nothing there in the spec is required

Reply:
So yes, this does sound optional, agreed

Reply:
I have to check if the same as here applies: https://github.com/kubernetes-sigs/cluster-api/pull/12431#discussion_r2179679815 (that at least an empty spec.template.spec is needed for the controller)

Reply:
Verified, it's optional. Probably it wouldn't work if we wouldn't default format because the topology controller probably needs .spec.template.spec